src/main/java/beam/analysis/plots/DeadHeadingAnalysis.java

```diff
@@ -151,7 +151,7 @@ private void processDeadHeadingDistanceRemainingRepositionings() {
                     for (int i = 0; i < m; i++) {
 
                         Event oldEvent = vehicleHourData.get(i);
-                        Double length2 = Double.parseDouble(oldEvent.getAttributes().get(PathTraversalEvent.ATTRIBUTE_LENGTH));
+                        Double length2 = ((PathTraversalEvent)oldEvent).legLength();
```

Comment:
Are we certain that this will always be of this type? If so then why not change the root to return that. Casting is unsafe, and especially without a type check first.

Reply:
I'm not sure for the future cases, but right now it will be only `PathTraversalEvent`. That map is modified in https://github.com/LBNL-UCB-STI/beam/blob/2e25cf3d430bdc48bf35e207a282af5306133397/src/main/java/beam/analysis/plots/DeadHeadingAnalysis.java#L165 which accept only `PathTraversalEvent`. Maybe I have to go and change the types for `vehicleEvents ` and `vehicleEventsCache `

Reply:
Can we do a type check then? Again, I'd rather catch potential misuse at compile time rather than runtime - as there's a chance we don't catch it until much later and tracking it down is harder...